src/Streamly/LZ4.hs

```diff
@@ -0,0 +1,106 @@
+-- |
+-- Module      : Streamly.LZ4
+-- Copyright   : (c) 2020 Composewell Technologies
+--
+-- License     : Apache-2.0
+-- Maintainer  : streamly@composewell.com
+-- Stability   : experimental
+-- Portability : GHC
+--
+-- This module provides streamly combinators that perform LZ4
+-- compression. Please take a look at <https://github.com/lz4/lz4> to learn
+-- more. All combinators work with stream of 'Array' provided by streamly.
+--
+-- The important combinators are 'compress' and 'decompress'. Their behaviour is
+-- straightforward.
+--
+module Streamly.LZ4
+    ( compress
+    , resize
+    , decompressResized
```

Comment:
We can remove `resize` and `decompressResized` from the export list. If you really want them to be available we can expose them from the Internal module.